shell/platform/android/io/flutter/embedding/android/FlutterActivityAndFragmentDelegate.java

```diff
@@ -444,11 +458,19 @@ void onDestroyView() {
   }
 
   void onSaveInstanceState(@Nullable Bundle bundle) {
-    Log.v(TAG, "onSaveInstanceState. Giving plugins an opportunity to save state.");
+    Log.v(TAG, "onSaveInstanceState. Giving framework and plugins an opportunity to save state.");
     ensureAlive();
 
+    if (host.shouldRestoreAndSaveState()) {
+      bundle.putByteArray(
+          FRAMEWORK_RESTORATION_BUNDLE_KEY,
+          flutterEngine.getRestorationChannel().getRestorationData());
+    }
```

Comment:
I don't really remember. We may have talked about this. Can't we clear the data now that we read this?

Reply:
Can you elaborate with a specific example? Perhaps write this down somewhere here for future archeologists too. I get the impression that we'll never run into a case where framework writes, engine reads and deletes, then the framework tries to read again? Is that right?

Reply:
I gave a longer answer to this on the other comment thread. In reality, the sequence of events that you describe will of course not happen. However, in reality, the only moment where we could delete the engine's copy of the restoration state is after we have handed it to the operating system. At that point, our app is moments away from getting killed anyways (thus freeing up all the memory). So why add the extra complexity of allowing framework and engine restoration state to go out of sync in this situation?